File: Golang/1979-find-greatest-common-divisor-of-array/find-greatest-common-divisor-of-array.go
func findGCD(nums []int) int {
    small, large := nums[0], nums[0]
    for i := 1; i < len(nums); i++ {
        if nums[i] < small {
            small = nums[i]
        }
        if nums[i] > large {
            large = nums[i]
        }
    }
    for i := small; i >= 1; i-- {
        if large % i == 0 && small % i == 0 {
            return i
        }
    }
    return 1
}

File: Python/2106-find-greatest-common-divisor-of-array/find-greatest-common-divisor-of-array.py
class Solution:
    def findGCD(self, nums: List[int]) -> int:
        large = small = nums[0]
        gcd = 1
        for num in nums:
            if num > large:
                large = num
            if num < small:
                small = num
        for i in range(small, 1, -1):
            if small % i == 0 and large % i == 0:
                gcd = i
                break
        return gcd